test/integration/cls-hooked/cls-hooked.test.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+const chai      = require('chai'),
+  expect    = chai.expect,
+  Support   = require('../support'),
+  Sequelize = Support.Sequelize,
+  Promise   = Sequelize.Promise,
+  cls       = require('cls-hooked'),
+  current = Support.sequelize,
+  semver = require('semver'),
+  isNode8 = semver.satisfies(process.version, '>= 8');
+
+if (current.dialect.supports.transactions && isNode8) {
+  describe(Support.getTestDialectTeaser('cls-hooked'), () => {
+    before(function() {
+      this.thenOriginal = Promise.prototype.then;
+      Sequelize.useCLS(cls.createNamespace('sequelize'));
+    });
+
+    after(() => {
+      delete Sequelize._cls;
+    });
+
+    let sequelize, ns, User;
+    beforeEach(async function() {
```

Comment:
Please don't use es7 :). Even though you have a guard at the start of the file, older versions will still try to parse the file.

This is copy-pasta of the current cls tests right? In that case, just replace the import of `continuation-local-storage` with `cls-hooked`. As far a I can see it should work on older versions as well

Reply:
Yes it's copy-paste. I can remove that, but the main reason why I added this was to see if the `async/await` features worked correctly. I realize that you have to support multiple versions of node, but how else could I test `async/await`? Do you think using `global.Promise` in the test cases instead of patched Bluebird would make a good-enough test case?

Reply:
If all tests pass on old version of node with cls hooked, I'm happy.

Async / await support is built into async_hooks, so we don't need to specifically test for it